ruby/lib/cucumber/ci_environment/variable_expression.rb

```diff
@@ -1,37 +1,42 @@
+# frozen_string_literal: true
+
 module Cucumber
   module CiEnvironment
     module VariableExpression
       def evaluate(expression, env)
         return nil if expression.nil?
+
         begin
           expression.gsub(/\${(.*?)(?:(?<!\\)\/(.*)\/(.*))?}/) do
-            variable = $1
-            pattern = $2
-            replacement = $3
+            variable = ::Regexp.last_match(1)
+            pattern = ::Regexp.last_match(2)
+            replacement = ::Regexp.last_match(3)
 
             value = get_value(variable, env)
             raise "Undefined variable #{variable}" if value.nil?
+
             if pattern.nil?
               value
             else
               regexp = Regexp.new(pattern.gsub('\/', '/'))
               match = value.match(regexp)
               raise "No match for variable #{variable}" if match.nil?
-              match[1..-1].each_with_index do |group, i|
-                replacement = replacement.gsub("\\#{i+1}", group)
+
+              match[1..].each_with_index do |group, i|
+                replacement = replacement.gsub("\\#{i + 1}", group)
               end
               replacement
             end
           end
-        rescue
+        rescue StandardError
```

Comment:
Lets leave this handling one out for now. It's fine not to fix everything.

This entire method is doing lots of skullduggery and we probably should be refactoring this to do something a bit neater